components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -2156,4 +2104,32 @@ private String getTenantUrl(String url, String tenantDomain) throws URISyntaxExc
         return uriModified.toString();
     }
 
+    private void verifyAndUpdateRoleConfiguration(String tenantDomain, int tenantId,
+            PermissionsAndRoleConfig roleConfiguration) throws IdentityProviderManagementException {
+
+        List<RoleMapping> validRoleMappings = new ArrayList<>();
+        List<String> validIdPRoles = new ArrayList<>();
+
+        for (RoleMapping mapping : roleConfiguration.getRoleMappings()) {
+            try {
+                UserStoreManager usm = IdPManagementServiceComponent.getRealmService().getTenantUserRealm(tenantId)
+                        .getUserStoreManager();
+                String role = mapping.getLocalRole().getLocalRoleName();
```

Comment:
Is there a possibility mapping.getLocalRole() can be null?

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/2155